http-client-jdk/src/main/java/io/micronaut/http/client/jdk/ByteBodySubscriber.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.jdk;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.http.body.CloseableByteBody;
+import io.micronaut.http.body.ReactiveByteBufferByteBody;
+import io.micronaut.http.body.stream.BodySizeLimits;
+import io.micronaut.http.body.stream.BufferConsumer;
+
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * {@link HttpResponse.BodySubscriber} implementation that pushes data into a
+ * {@link ReactiveByteBufferByteBody.SharedBuffer}.
+ *
+ * @since 4.8.0
+ * @author Jonas Konrad
+ */
+@Internal
+final class ByteBodySubscriber implements HttpResponse.BodySubscriber<CloseableByteBody>, BufferConsumer.Upstream {
```

Comment:
FYI: this has simply been moved from ReactiveByteBufferByteBody where it was an inner class previously, it's not new code.